### PR TITLE
caffeine-ng: 3.5.1 -> 4.0.2

### PR DESCRIPTION
**Pull request by SuperSamus**
###### Description of changes

Done some cleanup, and updated `caffeine-ng` to the new shiny version.
https://codeberg.org/WhyNotHugo/caffeine-ng/src/branch/main/CHANGELOG.rst

I only did the bare minimum of testing.

(It may not be important, but why does it print `No property specified.` every time it's enabled/disabled?)

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/X11/caffeine-ng/default.nix [hunk 1]
```diff
@@ -1,53 +1,90 @@
...
-    mkdir -p $out/share
     cp -r share $out/
-    cp -r caffeine/assets/icons $out/share/icons
+    cp -r caffeine/assets/icons $out/share/
+
     # autostart file
-    cp -r $out/lib/python*/site-packages/etc $out/etc/
+    ln -s $out/${python3.sitePackages}/etc $out/etc
+
     glib-compile-schemas --strict $out/share/glib-2.0/schemas
```
Comment (SuperSamus) [inline pkgs/tools/X11/caffeine-ng/default.nix:76]: Not very knowledgeable of Gnome, Is this important to have?

Comment (marzipankaiser) [reply pkgs/tools/X11/caffeine-ng/default.nix:76]: Me neither, but I think that it didn't work (complaining about not finding some schemas) without this when I originally packaged it.

Comment (marzipankaiser) [reply pkgs/tools/X11/caffeine-ng/default.nix:76]: Just tested it and this is still the case, without this line, I get the following error (and the program crashes):
```
(.caffeine-wrapped:105816): GLib-GIO-ERROR **: 14:06:20.475: Settings schema 'net.launchpad.caffeine' is not installed
```

File: pkgs/tools/X11/caffeine-ng/default.nix [hunk 2]
```diff
@@ -1,53 +1,90 @@
...
 
   doCheck = false; # There are no tests.
 
-  postPatch = ''
-    substituteInPlace caffeine/inhibitors.py \
-      --replace 'os.system("xset' 'os.system("${xset}/bin/xset' \
-      --replace 'os.system("xautolock' 'os.system("${xautolock}/bin/xautolock' \
-      --replace 'os.system("pgrep' 'os.system("${procps}/bin/pgrep' \
-      --replace 'os.system("xscreensaver-command' 'os.system("${xscreensaver}/bin/xscreensaver-command'
-  '';
```
Comment (marzipankaiser) [inline pkgs/tools/X11/caffeine-ng/default.nix:32]: Did you have a special reason for removing this?
On my machine, this makes e.g. `xautolock` lock the screen despite `caffeine-ng` being enabled, if `xautolock` is not in the `$PATH` for `caffeine-ng` (I haven't tested it with `xautolock` installed globally).

**Review by marzipankaiser — CHANGES_REQUESTED** [review]
Thanks for updating this.

I personally would prefer to still patch the paths for `xset` etc.
Does this pull in `xset` etc as dependencies? That would be unfortunate. Then again, having `caffeine-ng` "just work" without having to install tools globally or explicitly adding them to the path for `caffeine-ng` is nice and leads to the expected behavior in all cases.

**Comment by marzipankaiser** [comment]
As for the `No property specified.` - I don't know, but it does not seem to break anything important.

File: pkgs/tools/X11/caffeine-ng/default.nix (hunk 2 above)
Comment (SuperSamus) [reply pkgs/tools/X11/caffeine-ng/default.nix:32]: I did add them in `$PATH` in `gappsWrapperArgs`.

Comment (marzipankaiser) [reply pkgs/tools/X11/caffeine-ng/default.nix:32]: Ah, that's even better (and works too).

**Review by marzipankaiser — APPROVED** [review]

File: pkgs/tools/X11/caffeine-ng/default.nix [hunk 3]
```diff
@@ -1,53 +1,90 @@
...
   nativeBuildInputs = [ wrapGAppsHook glib ];
+
   buildInputs = [
-    gdk-pixbuf gobject-introspection libnotify gtk3
-    python3Packages.setuptools-scm
+    libappindicator-gtk3
+    libnotify
+    gobject-introspection
+    gtk3
+    setuptools-scm
```
Comment (SuperSandro2000) [inline pkgs/tools/X11/caffeine-ng/default.nix:54]: that goes into nativeBuildInputs

File: pkgs/tools/X11/caffeine-ng/default.nix [hunk 4]
```diff
@@ -1,53 +1,90 @@
...
+, pyxdg
+, setproctitle
+
+, python3
+, procps
+, xset
+, xautolock
+, xscreensaver
+, xfce
+
```
Comment (SuperSandro2000) [inline pkgs/tools/X11/caffeine-ng/default.nix:25]:
```suggestion
, gobject-introspection
, gtk3
, lib
, libappindicator-gtk3
, libnotify
, setuptools-scm
, click
, dbus-python
, ewmh
, pulsectl
, pygobject3
, pyxdg
, setproctitle
, python3
, procps
, xset
, xautolock
, xscreensaver
, xfce
```

**Pull request merged**